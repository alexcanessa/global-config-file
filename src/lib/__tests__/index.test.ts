import path from "path";
import globalConfigFile, { defaultExtensions } from "../";

const root = path.relative(process.cwd(), __dirname);
const deactivatedExtensions = Object.keys(defaultExtensions).reduce(
  (carry, extension) => ({
    ...carry,
    [extension]: false
  }),
  {}
);

describe("globalConfigFile function", () => {
  it("loads a .json file", () => {
    const config = globalConfigFile(".examplerc", {
      extensions: { ...deactivatedExtensions, json: true },
      root
    });

    expect(config).toMatchSnapshot();
  });
  it("loads a .yml file", () => {
    const config = globalConfigFile(".examplerc", {
      extensions: { ...deactivatedExtensions, yml: true },
      root
    });

    expect(config).toMatchSnapshot();
  });
  it("loads a .yaml file", () => {
    const config = globalConfigFile(".examplerc", {
      extensions: { ...deactivatedExtensions, yaml: true },
      root
    });

    expect(config).toMatchSnapshot();
  });
  it("loads an extension-free file", () => {
    const config = globalConfigFile(".examplerc", {
      extensions: { ...deactivatedExtensions, empty: true },
      root
    });

    expect(config).toMatchSnapshot();
  });
  it("loads a .js file", () => {
    const config = globalConfigFile(".examplerc", {
      extensions: { ...deactivatedExtensions, js: true },
      root
    });

    expect(config).toMatchSnapshot();
  });
  it("loads a .config.js file", () => {
    const config = globalConfigFile(".examplerc", {
      extensions: { ...deactivatedExtensions, "config.js": true },
      root
    });

    expect(config).toMatchSnapshot();
  });
  it("loads the package.json", () => {
    const config = globalConfigFile(".examplerc", {
      extensions: { ...deactivatedExtensions, package: true },
      root
    });

    expect(config).toMatchSnapshot();
  });
  it("throws an error if no files are found", () => {
    const globalConfigFileTest = globalConfigFile.bind(
      null,
      "nonExistingFile",
      {
        extensions: { ...deactivatedExtensions, json: true },
        root
      }
    );

    expect(globalConfigFileTest).toThrowErrorMatchingSnapshot();
  });
  it("throws an error if all falsy extensions are passed to options", () => {
    const globalConfigFileTest = globalConfigFile.bind(null, ".examplerc", {
      extensions: deactivatedExtensions,
      root
    });

    expect(globalConfigFileTest).toThrowErrorMatchingSnapshot();
  });
  it("does not throw an error if no files are found when quiet=true", () => {
    const globalConfigFileTest = globalConfigFile.bind(
      null,
      "nonExistingFile",
      {
        extensions: { ...deactivatedExtensions, json: true },
        root,
        quiet: true
      }
    );

    expect(globalConfigFileTest).not.toThrow();
  });
  it("does not throw an error if all falsy extensions are passed to options when quiet=true", () => {
    const globalConfigFileTest = globalConfigFile.bind(null, ".examplerc", {
      extensions: deactivatedExtensions,
      root,
      quiet: true
    });

    expect(globalConfigFileTest).not.toThrow();
  });
});
