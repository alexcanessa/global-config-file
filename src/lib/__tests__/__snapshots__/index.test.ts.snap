// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`globalConfigFile function loads a .config.js file 1`] = `
Object {
  "type": "config.js",
}
`;

exports[`globalConfigFile function loads a .js file 1`] = `
Object {
  "type": "js",
}
`;

exports[`globalConfigFile function loads a .json file 1`] = `
Object {
  "type": "json",
}
`;

exports[`globalConfigFile function loads a .yaml file 1`] = `
Object {
  "type": "yaml",
}
`;

exports[`globalConfigFile function loads a .yml file 1`] = `
Object {
  "type": "yml",
}
`;

exports[`globalConfigFile function loads an extension-free file 1`] = `
Object {
  "type": "raw, json",
}
`;

exports[`globalConfigFile function loads the package.json 1`] = `
Object {
  "type": "package.json",
}
`;

exports[`globalConfigFile function throws an error if all falsy extensions are passed to options 1`] = `"No extensions were found or enabled"`;

exports[`globalConfigFile function throws an error if no files are found 1`] = `"No configuration file found"`;
